core/src/main/java/org/elasticsearch/index/translog/TranslogWriter.java

```diff
@@ -80,6 +85,8 @@ public TranslogWriter(
         this.outputStream = new BufferedChannelOutputStream(java.nio.channels.Channels.newOutputStream(channel), bufferSize.bytesAsInt());
         this.lastSyncedCheckpoint = initialCheckpoint;
         this.totalOffset = initialCheckpoint.offset;
+        this.minSeqNo = initialCheckpoint.minSeqNo;
```

Comment:
I think this is right but it is confusing me some. Maybe you can explain it or maybe add a comment?

So my problem is that on figure reading it looks like you are keeping the minimum sequence number as made by some previous writer. So if you make more than one of these writers then this minimum value will stay the minimum value from the first writer. Would it make more sense to use the `NO_OPS_PERFORMED` constant here? Or maybe add a comment about what it when why we want this "dragging backwards" behavior sometimes.